libminifi/src/processors/FocusArchiveEntry.cpp

```diff
@@ -0,0 +1,340 @@
+
+      if (entryMetadata.entryType == AE_IFREG) {
+          entryVal["stash_key"] = Json::Value(entryMetadata.stashKey);
+      }
+
+      structVal.append(entryVal);
+    }
+
+    std::string archivenameStr;
+    Json::Value archiveName {Json::nullValue};
+
+    if (flowFile->getAttribute("filename", archivenameStr)) {
+       archiveName = Json::Value(archivenameStr);
+    }
+
+    Json::Value lensVal(Json::objectValue);
+    lensVal["archive_format_name"] = Json::Value(archiveMetadata.archiveFormatName);
+    lensVal["archive_name"] = archiveName;
+    lensVal["focused_entry"] = Json::Value(archiveMetadata.focusedEntry);
+    lensVal["archive_format"] = Json::Value(archiveMetadata.archiveFormat);
+    lensVal["archive_structure"] = structVal;
+    lensStack.append(lensVal);
+
+    Json::FastWriter writer;
+    std::string stackStr = writer.write(lensStack);
+
+    if (!flowFile->updateAttribute("lens.archive.stack", stackStr)) {
+      flowFile->addAttribute("lens.archive.stack", stackStr);
+    }
+  }
+
+  // Update filename attribute to that of focused entry
+  std::size_t found = targetEntry.find_last_of("/\\");
+  std::string path = targetEntry.substr(0, found);
+  std::string name = targetEntry.substr(found + 1);
+  set_del_or_update_attr(flowFile, "filename", &name);
+  set_del_or_update_attr(flowFile, "path", &path);
+  set_del_or_update_attr(flowFile, "absolute.path", &targetEntry);
+
+  // Transfer to the relationship
+  session->transfer(flowFile, Success);
+}
+
+typedef struct {
+  std::shared_ptr<io::BaseStream> stream;
+  char buf[8196];
+} FocusArchiveEntryReadData;
+
+int64_t FocusArchiveEntry::ReadCallback::process(std::shared_ptr<io::BaseStream> stream) {
+  auto inputArchive = archive_read_new();
+  struct archive_entry *entry;
+  int64_t nlen = 0;
+
+  FocusArchiveEntryReadData data;
+  data.stream = stream;
+
+  archive_read_support_format_all(inputArchive);
+  archive_read_support_filter_all(inputArchive);
+
+  // Read callback which reads from ifstream
+  auto read = [] (archive *, void *d, const void **buf) -> int64_t {
+    auto data = static_cast<FocusArchiveEntryReadData *>(d);
+    *buf = data->buf;
+    int64_t read = 0;
+    int64_t last_read = 0;
+
+    do {
+      last_read = data->stream->readData(reinterpret_cast<uint8_t *>(data->buf), 8196 - read);
+      read += last_read;
+    } while (last_read > 0 && read < 8196);
+
+    return read;
+  };
+
+  // Close callback for libarchive
+  auto close = [] (archive *, void *) -> int {
+    // Because we do not need to close the stream, do nothing & return success
+    return 0;
+  };
+
+  // Read each item in the archive
+  int res;
+
+  if ((res = archive_read_open(inputArchive, &data, NULL, read, close))) {
+      logger_->log_error(
+          "FocusArchiveEntry can't open due to archive error: %s",
+          archive_error_string(inputArchive));
+      return nlen;
+  }
+
+  for (;;) {
```

Comment:
^Same as above...isRunning() ?

Reply:
Fixed in d2e7e34ab8b331ac484b9b16bd51455799a1502b